File: src/main/java/com/hileco/cortex/instructions/bits/BITWISE_OR.java
package com.hileco.cortex.instructions.bits;

import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
public class BITWISE_OR extends BitInstruction {
    @Override
    public byte innerExecute(byte left, byte right) {
        var result = left;
        result |= right;
        return result;
    }
}

File: src/main/java/com/hileco/cortex/instructions/bits/BITWISE_OR.kt
package com.hileco.cortex.instructions.bits

import com.hileco.cortex.constraints.expressions.Expression
import kotlin.experimental.or

class BITWISE_OR : BitInstruction() {
    override fun innerExecute(left: Byte, right: Byte): Byte {
        return left or right
    }

    override fun innerExecute(left: Expression, right: Expression): Expression {
        throw UnsupportedOperationException()
    }
}